src/cpp/handler/handlermain.cpp

```diff
@@ -24,18 +24,15 @@
 #include <QTimer>
 #include "handlerapp.h"
 
-class HandlerAppMain : public QObject
+class HandlerAppMain
 {
-	Q_OBJECT
-
 public:
 	HandlerApp *app;
 
-public slots:
 	void start()
 	{
-		app = new HandlerApp(this);
```

Comment:
one drawback to not using QObject is we don't get automatic cleanup. before, this parenting would automatically destroy the `HandlerApp` when the `HandlerAppMain` itself was destroyed. probably we'll want to move to some other non-Qt solution for this, but in the meantime lets just clean things up manually with a destructor. we should also disconnect the signal there too, prior to destruction, e.g.:

```c++
~HandlerAppMain()
{
        quitConnection.disconnect();
        delete app;
}
```

Reply:
`quicConnection.disconnect()` & `hupConnection.disconnect()` can be added to `~HandlerApp()` 
and then maybe call `~HandlerApp()` from within `~HandlerAppMain()` ?